### ISSUE TITLE
QuickEdit UI for Padding / Margin

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#2507]
It would be great to have a QuickEdit UI (similar to how we handle the colorpicker) for the CSS properties ``padding`` & ``margin``. The goal would be to create an expandable UI that would allow people to set these properties (they have a lot of similarities, so this would be easy to re-use).

Here's a very rough mockup of the UI controls it could have...

```
(*) Set individual margin (or padding) for each side
( ) Use same value for all sides

Units: 
(*) px
( ) em
( ) %

Top:    -----O------ 5px
Right:  ---------O-- 7px
Bottom: O----------- 8px
Left:   --O--------- 2px
```

The UI would be a combination of toggles and sliders and it would update the code in the editor to match what the user selected.

If we can get these basic controls working, then we can move to a secondary phase of improving the look and feel of the UI.

**Comment by bansalnitish** [comment]
May I give it a try ?

**Comment by humphd** [comment]
@bansalnitish this inline editors are big, but you're welcome to do it if you'd like.  Here's a recent example that got added:

https://github.com/mozilla/brackets/commit/127fc1736dcf83c53868539a98f1b1399e55321d

Personally, I'd suggest people do other things.  But I'll let you decide.

**Comment by bansalnitish** [comment]
I'll follow the example and try this one out !

**Comment by humphd** [comment]
OK, sounds good. There are some other examples of inline editors in the extensions dir as well.

**Comment by bansalnitish** [comment]
@humphd I have made the PR for `padding`. Should I continue in the same way for `margin` as well ?

**Comment by humphd** [comment]
Sure, doing `margin` would be cool.  Question: how much of this current code could you share between them?  It would be good if we didn't have to make a ton of new code for everything like this, especially if they share basic things.

**Comment by bansalnitish** [comment]
I'll just try to share as much as I can and then we'll discuss how to proceed further !

**Comment by gideonthomas** [comment]
Fixed with mozilla/brackets#916
